drivers/i2c/Kconfig.nrfx

```diff
@@ -0,0 +1,119 @@
+# Kconfig - nrfx I2C support
+#
+# Copyright (c) 2018, Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig I2C_NRFX
+	bool "nRF TWI nrfx drivers"
+	depends on SOC_FAMILY_NRF
+	help
+	  Enable support for nrfx TWI drivers for nRF MCU series.
+	  Peripherals with the same instance ID cannot be used together,
+	  e.g. I2C_0 and SPI_0. You may need to disable SPI_0 or SPI_1.
+
+if I2C_NRFX
+
+config I2C_NRF_OVERRIDE_DTS_PINS
```

Comment:
Why do we need this?

Reply:
It is mainly used for development kits, where I2C pins can be changed quite often, depending on hardware connected to the board.
We can't use configuration from DTS as default pins for Kconfig, but by adding override option we can switch to using pin configuration from DTS or to using user configuration from menuconfig.

Reply:
> It is mainly used for development kits, where I2C pins can be changed quite often, depending on hardware connected to the board.
> We can't use configuration from DTS as default pins for Kconfig, but by adding override option we can switch to using pin configuration from DTS or to using user configuration from menuconfig.

I'm not keen on having support for this, we should be doing HW cfg via the DTS and only DTS.

Reply:
Actually since there is an active discussion regarding Kconfig and DT I'll copy a couple of people here who have been involved so we get more perspectives: @erwango @tbursztyka @SebastianBoe

Reply:
think having multiple ways for HW configuration (or anything in general) is not a good idea.
Some will use one way, some will use other, it'll soon become difficult to maintain.

Reply:
Aren't sda/scd pins the same on 1 board, no matter how many slaves you wire up?

Reply:
> Aren't sda/scd pins the same on 1 board, no matter how many slaves you wire up?

No, in NRF SoCs you can connect SDA/SDC PINs wherevere you wish and you can modify them during runtime.
I saw that for instance ST can have 2 sets of PINs for UART in DTS. We can be more flexible thus we would like to have  possibility to overwride DTS in Kconfig. Putting all possible combinations to DTS makes limited sense in my opinion.